src/realm/mixed.cpp

```diff
@@ -372,27 +372,66 @@ int Mixed::compare_signed(const Mixed& b) const
     return compare(b);
 }
 
-template <class T>
-T Mixed::export_to_type() const noexcept
+template <>
+int64_t Mixed::export_to_type() const noexcept
+{
+    REALM_ASSERT(get_type() == type_Int);
+    return int_val;
+}
+
+template <>
+float Mixed::export_to_type() const noexcept
 {
     REALM_ASSERT(m_type);
     switch (get_type()) {
         case type_Int:
-            return T(int_val);
+            return float(int_val);
         case type_Float:
-            return T(float_val);
+            return float_val;
+        default:
+            REALM_ASSERT(false);
+            break;
+    }
+    return 0.;
+}
+
+template <>
+double Mixed::export_to_type() const noexcept
+{
+    REALM_ASSERT(m_type);
+    switch (get_type()) {
+        case type_Int:
+            return double(int_val);
+        case type_Float:
+            return double(float_val);
         case type_Double:
-            return T(double_val);
+            return double_val;
         default:
             REALM_ASSERT(false);
             break;
     }
-    return T();
+    return 0.;
 }
 
-template int64_t Mixed::export_to_type<int64_t>() const noexcept;
-template float Mixed::export_to_type<float>() const noexcept;
-template double Mixed::export_to_type<double>() const noexcept;
+template <>
+Decimal128 Mixed::export_to_type() const noexcept
+{
+    REALM_ASSERT(m_type);
+    switch (get_type()) {
+        case type_Int:
+            return Decimal128(int_val);
+        case type_Float:
+            return Decimal128(float_val);
+        case type_Double:
+            return Decimal128(double_val);
+        case type_Decimal:
+            return decimal_val;
+        default:
+            REALM_ASSERT(false);
```

Comment:
I think these assertions are going to be a problem when comparing two mixed properties when one value is a decimal and the other is some non-numeric type?

Reply:
I now have a check in the Parser that the involved types are numeric.